presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -370,4 +375,28 @@ public void testCorrelatedScalarAggregationRewriteToLeftOuterJoin()
                                                                                 project(ImmutableMap.of("NON_NULL", expression("true")),
                                                                                         node(ValuesNode.class)))))))))));
     }
+    @Test
+    public void testCountStarOverInnerAggregateQuery()
+    {
+        assertPlan(
+                "SELECT count(*) from (select sum(orderkey) from orders)",
+                output(node(ValuesNode.class).with(new ValuesMatcher(ImmutableMap.of(), Optional.of(Integer.valueOf(1)), Optional.of(ImmutableList.of(ImmutableList.of(new LongLiteral("1"))))))));
+    }
+    @Test
+    public void testCountColumnOverInnerAggregateQuery()
+    {
+        assertPlan(
+                "SELECT count(s) from (select sum(orderkey) as s from orders)",
+                output(aggregation(ImmutableMap.of("COUNT", functionCall("count", ImmutableList.of("FINAL_SUM"))),
+                            aggregation(ImmutableMap.of("FINAL_SUM", functionCall("sum", ImmutableList.of("PARTIAL_SUM"))),
+                                    exchange(ExchangeNode.Scope.LOCAL, ExchangeNode.Type.GATHER,
+                                            aggregation(ImmutableMap.of("PARTIAL_SUM", functionCall("sum", ImmutableList.of("ORDERKEY"))),
+                                                    tableScan("orders", ImmutableMap.of("ORDERKEY", "orderkey"))
```

Comment:
I would just use anyTreeNode with tableScan as child. That would mean exactly that plan was not optimized.

```
 assertPlan(
                 "SELECT count(s) from (select sum(orderkey) as s from orders)",
                 anyTreeNode(
                     tableScan("orders')));
```